src/backend/tcop/postgres.c

```diff
@@ -1560,16 +1561,23 @@ CheckDebugDtmActionSqlCommandTag(const char *sqlCommandTag)
 }
 
 static void
-restore_guc_to_QE(void )
+send_guc_to_QE(List *guc_list, bool is_restore)
 {
-	Assert(Gp_role == GP_ROLE_DISPATCH && gp_guc_restore_list);
+	Assert(Gp_role == GP_ROLE_DISPATCH && guc_list);
 	ListCell *lc;
+	MemoryContext oldcontext = CurrentMemoryContext;
 
 	start_xact_command();
 
-	foreach(lc, gp_guc_restore_list)
+	foreach(lc, guc_list)
 	{
 		struct config_generic* gconfig = (struct config_generic *)lfirst(lc);
+
+		/* We can't SET GUCs whose context is too high. */
```

Comment:
I think comment should be updated.

Reply:
To what? It's still relevant.

Reply:
this comment is only about condition `gconfig->context < PGC_SUSET`